src/components/erc721/common/declining_price_graph.tsx

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import { Area, AreaChart, ResponsiveContainer, Tooltip, YAxis } from 'recharts';
+
+interface Props {
+    beginAmountInUnits: number;
+    endAmountInUnits: number;
+    currentPriceInUnits: number;
+}
+
+const CustomTooltip = (props: any) => {
+    const { active } = props;
+
+    if (active) {
+        const { payload } = props;
+        return <span>{payload[0].value} ETH</span>;
+    }
+
+    return null;
+};
+
+class DecliningPriceGraph extends React.Component<Props> {
+    public state: any = {
+        dataNumbers: [],
+    };
+
+    public componentDidMount = () => {
+        this._generateMarkersWithPrices();
+    };
+
+    public componentDidUpdate = (prevProps: Readonly<Props>) => {
+        const { currentPriceInUnits } = this.props;
+        if (prevProps.currentPriceInUnits !== currentPriceInUnits) {
+            this._generateMarkersWithPrices();
+        }
+    };
+
+    public render = () => {
+        return (
+            <ResponsiveContainer width="100%" height={148}>
+                <AreaChart data={this.state.dataNumbers} margin={{ top: 15, right: 30, left: 20, bottom: 15 }}>
+                    <Tooltip content={CustomTooltip} />
+                    <YAxis
+                        hide={true}
+                        domain={[dataMin => this.props.endAmountInUnits, dataMax => this.props.beginAmountInUnits]}
+                    />
+                    <Area
+                        type="monotone"
+                        dataKey="price"
+                        stroke="#00AE99"
+                        fill="#00AE99"
+                        activeDot={{ r: 4 }}
+                        dot={false}
+                    />
+                    <Area
+                        type="monotone"
+                        dataKey="markerLimit"
+                        stroke="#00AE99"
+                        fill="#00AE99"
+                        dot={{ stroke: '#00AE99', strokeWidth: 4 }}
+                    />
+                    <Area
+                        type="monotone"
+                        dataKey="currentPrice"
+                        stroke="#00AE99"
+                        fill="#00AE99"
```

Comment:
R: Could you put is values in as constants?

Reply:
Added the color values to the theme properties